chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -364,12 +364,72 @@ where
         Ok(())
     }
 
+    fn check_stake_outputs_in_reward(
+        &self,
+        block: &WithId<Block>,
+    ) -> Result<(), ConnectTransactionError> {
+        match block.consensus_data() {
+            ConsensusData::None | ConsensusData::PoW(_) => Ok(()),
+            ConsensusData::PoS(_) => {
+                let block_reward_transactable = block.block_reward_transactable();
+
+                let kernel_input =
```

Comment:
Isn't there a function, get_kernel_output, that does this for you? Why repeating the logic here?